### PR TITLE
[WIP: just an example] make it work

### PR DESCRIPTION
**Pull request by domenicrosati**
This is mostly from Guillaumes PR with several fundemental differneces:
1.  The problematic nature of the expiry segment code 
2. Several issues with actually using `setAuthToken` 
3. the data structure of the profile request

### CODE REVIEW

File: src/utils/connections.ts [hunk 1]
```diff
@@ -18,6 +19,7 @@ export const connections: Connections = {
     marketplace: 'http://api.marketplace.arigato.tools/v1',
     provisioning: 'http://api.provisioning.arigato.tools/v1',
     connector: 'http://api.connector.arigato.tools/v1',
+    graphQl: 'http://graphql.arigato.tools/graphql',
```
Comment (jelmersnoeck) [inline src/utils/connections.ts]: Oh this is interesting. Seeing my other comment wrgt the puma URL for the component, would it be possible to put that in here too? And then it'll automatically load the correct one based on the env?

File: src/components/manifold-auth-token/manifold-auth-token.tsx [hunk 2]
```diff
@@ -26,9 +26,11 @@ export class ManifoldAuthToken {
 
   setInternalToken = (e: CustomEvent) => {
     const payload = e.detail as AuthToken;
+    console.log(payload);
     if (!payload.error && payload.expiry) {
       const expiry = payload.expiry.toString();
       this.token = `${payload.token}.${expiry}`;
+      console.log(this.token);
```
[the remark this answers is not in the document]
Comment (davidleger95) [reply src/components/manifold-auth-token/manifold-auth-token.tsx]: image

File: src/components/manifold-auth-token/manifold-auth-token.tsx [hunk 3]
```diff
@@ -37,7 +39,7 @@ export class ManifoldAuthToken {
       <div>
         <manifold-oauth
           onReceiveManifoldToken={this.setInternalToken}
-          oauthUrl="https://manifold-shadowcat-test-server.herokuapp.com/signin/oauth/web"
+          oauthUrl="http://login.arigato.tools/signin/oauth/web"
```
[the remark this answers is not in the document]
Comment (davidleger95) [reply src/components/manifold-auth-token/manifold-auth-token.tsx]: I just merged a PR yesterday that makes it customizable

File: src/components/manifold-connection/manifold-connection.tsx [hunk 4]
```diff
@@ -19,29 +19,29 @@ function getDefaultToken() {
...
   };
 
-  getAuthToken = () => this.authToken;
+  getAuthToken = () => this.accessToken;
 
   get accessToken() {
     if (this.authToken) {
-      const [token] = this.authToken.split('.');
+      // the auth token's last segment is its expiry
+      const token = this.authToken.split('.', this.authToken.split('.').length - 1).join('.');
```
Comment (davidleger95) [inline src/components/manifold-connection/manifold-connection.tsx]: we made a util function that gets the expiry. maybe we need one that gets the token too?

(Perhaps we can make a regex that separates these pieces for us as a long term solution? We'll see what the priorities are next week though)

File: src/utils/connections.ts (hunk 1 above)
Comment (davidleger95) [reply src/utils/connections.ts]: +1 on this

File: src/components/manifold-connection/manifold-connection.tsx (hunk 4 above)
Comment (Minivera) [reply src/components/manifold-connection/manifold-connection.tsx]: This PR predates that util function, which is why that change is necessary. Rebasing on master should do the trick.

File: src/components/manifold-data-provision-button/manifold-data-provision-button.tsx [hunk 5]
```diff
@@ -21,13 +23,31 @@ interface ErrorMessage {
   resourceLabel?: string;
 }
 
+interface Profile {
+  id: string;
+}
+
+interface ProfileMessage {
+  profile: Profile;
```
Comment (Minivera) [inline src/components/manifold-data-provision-button/manifold-data-provision-button.tsx]: 👍

File: src/utils/connections.ts (hunk 1 above)
Comment (dangodev) [reply src/utils/connections.ts]: Minor nitpick: can this be `graphQL` (`L` capitalized)? I’m open to `graphql`, `gql`, or `graphQL`, but IMO anything else is awkward

**Pull request closed without merging**